test/strategy/liquity/LiquityStrategy.spec.ts

```diff
@@ -557,17 +557,60 @@ describe('LiquityStrategy', () => {
       ).to.be.revertedWith('StrategyNothingToReinvest');
     });
 
-    it('fails if the minimum principal protection is not guaranteed', async () => {
-      await depositToVault(parseUnits('100'));
-      await strategy.setMinPrincipalProtectionPct(12000);
+    it('protects pricipal by reverting when total underlying assets are greater than min protected principal', async () => {
+      await strategy.setMinPrincipalProtectionPct('11000'); // 110%
+      await addUnderlyingBalance(alice, '10000');
+      await depositToVault(alice, '10000');
+      await vault.updateInvested();
 
-      await strategy.connect(admin).allowSwapTarget(SWAP_TARGET);
+      const ethAmount = parseUnits('2000');
+      await setBalance(strategy.address, ethAmount);
+
+      expect(await vault.totalUnderlying()).to.gt(parseUnits('11000'));
+
+      // amount out min + principal < min protected principal
+      const amountOutMin = parseUnits('500').sub(1);
 
       await expect(
-        strategy.connect(keeper).reinvest(SWAP_TARGET, 0, [], 0, [], 0),
+        strategy
+          .connect(keeper)
+          .reinvest(mock0x.address, 0, [], ethAmount, [], amountOutMin),
       ).to.be.revertedWith('StrategyMinimumPrincipalProtection');
     });
 
+    it('bypasses principal protection when total underlying assets are less than min protected principal', async () => {
+      await strategy.setMinPrincipalProtectionPct('12000'); // 120%
+      await addUnderlyingBalance(alice, '10000');
+      await depositToVault(alice, '10000');
+      await vault.updateInvested();
+
+      const ethAmount = parseUnits('1000');
+      await setBalance(strategy.address, ethAmount);
+      const ethSwapData = getSwapData(
+        constants.AddressZero,
+        underlying,
+        ethAmount,
+      );
+
+      expect(await vault.totalUnderlying()).to.lt(parseUnits('12000'));
```

Comment:
can we get a check for how much is in the vault and how much is in the strategy? it's a bit hard to keep track of because of the exchange rate from eth to lusd

Reply:
[fixed](https://github.com/lindy-labs/sc_solidity-contracts/pull/219/commits/5175c4b34070339e6feae953a5df246d169de353)